### ISSUE TITLE
[Popover] a11y review

### ISSUE THREAD

**Issue by Westbrook** [adobe/spectrum-web-components#1277]
### Feedback

- I’m not understanding what the dialog attribute is doing - it doesn’t appear to be documented in the API documentation
- I’d like to see examples which show real usage as I’m not clear if labeling and describing can work correctly in these examples

### Path forward

1. Add `dialog` to the element API/docs. In short, it's a decorative attribute that adds padding to the container when padding is not expected to be provided by the content.
2. _Generally_, popover should be presentational only... in that way it's used to ensure the visual delivery of things like `sp-picker` or other content that will be functionally powered by the `Overlay` API. In that way, I'm not 100% sure what content would be useful here. Let's discuss further.

cc @jnurthen

**Comment by Westbrook** [comment]
@jnurthen I'm addressing the documentation of the API in #1935 but I'm not sure how I'd specifically discuss usage here. _Mostly_, the `<sp-popover>` is a fancy DIV and isn't really part of the larger accessibility tree. In that way, an `<sp-dialog>` or `<sp-menu>`, or anything really, which manages it's own accessibility could be placed in it. With that in mind, would it benefit from having `role="presentation"` by default, or is that _implied_?
